exporter/skywalkingexporter/metricrecord_to_metricdata.go

```diff
@@ -0,0 +1,234 @@
+		svc.SingleValue.Labels = labels
+		meterDataCount.Timestamp = dataPoint.Timestamp().AsTime().UnixMilli()
+		svc.SingleValue.Name = name + "_count"
+		svc.SingleValue.Value = float64(dataPoint.Count())
+		meterDataCount.Metric = svc
+		meterDataCount.Service = defaultServiceName
+		meterDataCount.ServiceInstance = defaultServiceInstance
+		metrics = append(metrics, meterDataCount)
+	}
+	return metrics
+}
+
+func doubleSummaryMetricsToData(name string, data pdata.SummaryDataPointSlice, defaultLabels []*metricpb.Label) (metrics []*metricpb.MeterData) {
+	metrics = make([]*metricpb.MeterData, 0)
+	for i := 0; i < data.Len(); i++ {
+		dataPoint := data.At(i)
+		attributeMap := dataPoint.Attributes()
+		labels := make([]*metricpb.Label, 0)
+		attributeMap.Range(func(k string, v pdata.AttributeValue) bool {
+			labels = append(labels, &metricpb.Label{Name: k, Value: v.AsString()})
+			return true
+		})
+
+		for _, label := range defaultLabels {
+			labels = append(labels, &metricpb.Label{Name: label.Name, Value: label.Value})
+		}
+
+		values := dataPoint.QuantileValues()
+		for i := 0; i < values.Len(); i++ {
+			value := values.At(i)
+			meterData := &metricpb.MeterData{}
+			sv := &metricpb.MeterData_SingleValue{SingleValue: &metricpb.MeterSingleValue{}}
+			svLabels := make([]*metricpb.Label, 0)
+			svLabels = append(svLabels, labels...)
+			svLabels = append(svLabels, &metricpb.Label{Name: "quantile", Value: strconv.FormatFloat(value.Quantile(), 'g', -1, 64)})
+			sv.SingleValue.Labels = svLabels
+			meterData.Timestamp = dataPoint.Timestamp().AsTime().UnixMilli()
+			sv.SingleValue.Name = name
+			sv.SingleValue.Value = value.Value()
+			meterData.Metric = sv
+			meterData.Service = defaultServiceName
+			meterData.ServiceInstance = defaultServiceInstance
+			metrics = append(metrics, meterData)
+		}
+
+		meterDataSum := &metricpb.MeterData{}
+		svs := &metricpb.MeterData_SingleValue{SingleValue: &metricpb.MeterSingleValue{}}
+		svs.SingleValue.Labels = labels
+		svs.SingleValue.Name = name + "_sum"
+		svs.SingleValue.Value = dataPoint.Sum()
+		meterDataSum.Metric = svs
+		meterDataSum.Timestamp = dataPoint.Timestamp().AsTime().UnixMilli()
+		meterDataSum.Service = defaultServiceName
+		meterDataSum.ServiceInstance = defaultServiceInstance
+		metrics = append(metrics, meterDataSum)
+
+		meterDataCount := &metricpb.MeterData{}
+		svc := &metricpb.MeterData_SingleValue{SingleValue: &metricpb.MeterSingleValue{}}
+		svc.SingleValue.Labels = labels
+		meterDataCount.Timestamp = dataPoint.Timestamp().AsTime().UnixMilli()
+		svc.SingleValue.Name = name + "_count"
+		svc.SingleValue.Value = float64(dataPoint.Count())
+		meterDataCount.Metric = svc
+		meterDataCount.Service = defaultServiceName
+		meterDataCount.ServiceInstance = defaultServiceInstance
+		metrics = append(metrics, meterDataCount)
+	}
+	return metrics
+}
+
+func metricDataToMetricData(md pdata.Metric, defaultLabels []*metricpb.Label) (metrics []*metricpb.MeterData) {
```

Comment:
What about `MetricDataTypeExponentialHistogram`? It's not necessary to do it on this PR, and it's not a widely used type yet, but you may want to create an issue to add support for this type. 

Also, it's possible that OTLP will have new types over time. It may be a good idea to have some observability feature for the user to know when datapoints are dropped because a metric type is not supported by this exporter.

Reply:
Do you have any suggestions about observability.  Maybe Indicate the supported types in the document, or print the warn log when encountering types that cannot be processed. I see that the existing exporters basically ignore this.

Reply:
>  Maybe Indicate the supported types in the document, or print the warn log when encountering types that cannot be processed

Both seem like good things to have (maybe with a debug level log instead, to be non-spammy by default?). It's true that other exporters ignore this issue, but I think it's important to have *something* until you support `ExponentialHistogram`, so that end-users have some lead when understanding why they don't see the metrics in your backend.